usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllRequiredFilePresenceTest.java

```diff
@@ -17,220 +17,92 @@
+        Set<String> testSet = new HashSet<>(List.of("req0.req","req1.req","req2.req","req3.req","req4.req",
+                                                    "req5.req","req6.req","req7.req","req8.req","req9.req",
+                                                    "opt0.opt","opt1.opt","opt2.opt","opt3.opt","opt4.opt",
+                                                    "opt5.opt","opt6.opt","opt7.opt","opt8.opt","opt9.opt"));
+        when(mockFileRepo.getFilenameAll()).thenReturn(testSet);
+
+        GtfsSpecRepository mockSpecRepo = mock(GtfsSpecRepository.class);
+        List<String> testRequiredList = new ArrayList<>(List.of("req0.req","req1.req","req2.req","req3.req","req4.req",
+                "req5.req","req6.req","req7.req","req8.req","req9.req"));
+        when(mockSpecRepo.getRequiredFilenameList()).thenReturn(testRequiredList);
+
+        ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        when(mockResultRepo.addNotice(any(ErrorNotice.class))).thenReturn(null);
 
         ValidateAllRequiredFilePresence underTest = new ValidateAllRequiredFilePresence(
-                new MockSpecRepo(10),
-                new MockRawFileRepo(10, 10),
+                mockSpecRepo,
+                mockFileRepo,
                 mockResultRepo
-
         );
 
         List<String> result = underTest.execute();
-        assertEquals(0, mockResultRepo.notices.size());
         assertEquals(10, result.size());
         assertEquals(List.of("req0.req", "req1.req", "req2.req", "req3.req", "req4.req", "req5.req", "req6.req",
                 "req7.req", "req8.req", "req9.req"), result);
+
+        InOrder inOrder = Mockito.inOrder(mockFileRepo, mockSpecRepo);
+
+        inOrder.verify(mockFileRepo, times(1)).getFilenameAll();
+        inOrder.verify(mockSpecRepo, times(2)).getRequiredFilenameList();
+        verifyNoInteractions(mockResultRepo);
+        verifyNoMoreInteractions(mockFileRepo, mockSpecRepo, mockResultRepo);
     }
 
     @Test
     void missingRequiredShouldGenerateOneNoticePerMissingFile() {
 
-        MockValidationResultRepo mockResultRepo = new MockValidationResultRepo();
+        RawFileRepository mockFileRepo = mock(RawFileRepository.class);
+        Set<String> testSet = new HashSet<>(List.of("req0.req","req1.req","req2.req","req3.req","req4.req",
+                                                    "req5.req","req6.req","req7.req","req8.req","req9.req",
+                                                    "opt0.opt","opt1.opt","opt2.opt","opt3.opt","opt4.opt",
+                                                    "opt5.opt","opt6.opt","opt7.opt","opt8.opt","opt9.opt"));
+        when(mockFileRepo.getFilenameAll()).thenReturn(testSet);
+
+        GtfsSpecRepository mockSpecRepo = mock(GtfsSpecRepository.class);
+        List<String> testRequiredList = new ArrayList<>(List.of("req0.req","req1.req","req2.req","req3.req","req4.req",
```

Comment:
Can be simplified, directly affecting List.of

Reply:
Fixed